src/OrleansRuntime/Catalog/ActivationData.cs

```diff
@@ -321,6 +322,8 @@ public GrainId Identity
 
         public ActivationAddress Address { get; private set; }
 
+        public IServiceProvider ServiceProvider => this.ServiceScope?.ServiceProvider;
```

Comment:
similar to my other answer, I kept it this way so that you can still new up grains in your unit tests without using an actual Silo. Not sure if it's a good idea, but that PR was merged, so I assume people liked that option to not throw. I could make it throw if that's preferred